src/main/java/org/elasticsearch/index/mapper/internal/AllFieldMapper.java

```diff
@@ -155,8 +149,10 @@ public AllFieldMapper build(BuilderContext context) {
                     builder.enabled(nodeBooleanValue(fieldNode) ? EnabledAttributeMapper.ENABLED : EnabledAttributeMapper.DISABLED);
                     iterator.remove();
                 } else if (fieldName.equals("auto_boost")) {
-                    builder.autoBoost = nodeBooleanValue(fieldNode);
-                    iterator.remove();
+                    if (parserContext.indexVersionCreated().before(Version.V_2_0_0)) {
```

Comment:
join this into the else if? this way for 2.0 auto_boost will be rejected as an unknown setting.